src/app/model/ChessEngine.ts

```diff
@@ -99,8 +111,107 @@ export class ChessEngine implements ChessModel {
     }
   }
 
+  private intendToCastling(squareFrom: Square, squareTo: Square): boolean {
```

Comment:
Nie da się zrobić roszady:
![image](https://user-images.githubusercontent.com/18637116/108040930-34bcbe00-703e-11eb-993c-c189a869d169.png)
Trzeba rozszerzyć `possibleMoves` w ChessEngine o zwracanie możliwości roszady :) Podobnie jak jest obcinana mozliwosc wykonania ruchu, który szacuje króla - to też wpływa na possibleMoves, które to potem sa wyswietlane na UI.

Reply:
Nie rozumiem, przecież rozszerzyłam possibleMoves o te ruchy.

Reply:
Chyba, że chodzi o possibleMoves stricte dla króla, ale czy to jest dobre rozwiązanie? - nie wiem

Reply:
A nie, to nie dobre :) To czkaj... hmmm... ja tego nie widzę. Gdzie to jest :) ? 
Jest takie wymaganie:
```
Metoda zwracajaca mozliwe ruchy powinna w przypadku mozliwosci roszady - dodać do możliwych ruchów Króla także tą opcję. [mówimy o ChessEngine - przypisek redakcji :D]
```

Reply:
Teraz possibleMoves ma tylko to:
```
  public pieceMovesNotCausingAllyKingCheckmate(position: Square): Square[] {
    const initialPossibleMoves = this.board.onPositionPiece(position)?.possibleMoves(position, this.board) ?? [];
    const filteringFunction = (onePossibleMove: Square) => !this.willBeKingChecked(position, onePossibleMove);
    return initialPossibleMoves.filter(filteringFunction);
  }

  public possibleMoves(position: Square): Square[] {
    return this.pieceMovesNotCausingAllyKingCheckmate(position);
  }
```

i nie ma tych ruchów i nie możemy się ruszyć. Bo UI korzysta z tego. W logice nawet też mogłabyś skorzystać ewentualnie z tej samej metody :)

Reply:
` private canMoveOnSquare(squareFrom: Square, squareTo: Square): boolean {
    const piecePossibleMoves = this.board.onPositionPiece(squareFrom)?.possibleMoves(squareFrom, this.board);
    if (this.intendToCastling(squareFrom, squareTo) && this.isCastlingPossible(squareFrom, squareTo)) {
      this.castlingOnSquare = squareTo;
      piecePossibleMoves?.push(squareTo);`

Reply:
aa, ok

Reply:
No dobra, ale sprawdzasz przy wykonywaniu ruchów. I to jest ta pierwsza czesc zadania z movePiece.
A nie sprawdzasz przy zwracaniu mozliwych ruchow dla bierki. No sprawdz sobie, ze nie mozesz wykonac w grze roszady :)